opentelemetry-sdk/src/metrics/internal/exponential_histogram.rs

```diff
@@ -395,62 +391,39 @@ impl<T: Number<T>> ExpoHistogram<T> {
 
         let n = values.len();
         if n > h.data_points.capacity() {
-            h.data_points.reserve(n - h.data_points.capacity());
+            h.data_points.reserve_exact(n - h.data_points.capacity());
```

Comment:
Any reason to use `reserve_exact`. Without much context, you risk losing the amortized constant time of insertions, since you may get lots of small increments that makes the `Vec` have to reallocate every/most of the time.

Reply:
it's a small optimization, but the data point usage is a memory high water mark so this caps the memory usage at what you need. It only happens once per collection cycle, and only if your data point count is higher than the current capacity so it shouldn't be a collection perf imact in common cases.